backend/src/auth/depends.py

```diff
@@ -18,25 +20,48 @@ def verify(password, hashed_password):
 
 # Check whether the hash value of password is same as hashed_password
 def check_user_password_is_correct(database, name, password):
-    isPasswordCorrect = database.query(User).filter(User.username == name).first()
-    if not verify(password, isPasswordCorrect.hashed_password):
+    userdata = database.query(User).filter(User.username == name).first()
+    try:
+        if not verify(password, userdata.hashed_password):
+            return False
+    except Exception as e:
+        logging.error(f"Failed to verify password: {e}")
+        capture_exception(e)
         return False
-    return isPasswordCorrect
+    return userdata
 
 def authenticate_user_token(
     token = Depends(oauth2_scheme),
     database = Depends(session_opener)
 ):
-    payload = jwt.decode(token, Constants.KEY, algorithms=[Constants.ENCODING_ALGO])
+    try:
+        logging.debug(f"Authenticating token: {token[:10]}...")
+        payload = jwt.decode(token, Constants.Auth.KEY, algorithms=["HS256"])
+    except ExpiredSignatureError as e:
+        logging.error(f"Token expired: {e}")
+        capture_exception(e)
+        raise HTTPException(status_code=401, detail="Token expired")
+    except JWTError as e:
+        logging.error(f"Token invalid: {e}")
+        capture_exception(e)
+        raise HTTPException(status_code=401, detail="Invalid token")
+    except Exception as e:
+        logging.error(f"Failed to authenticate token: {e}")
+        capture_exception(e)
+        raise HTTPException(status_code=401, detail="Failed to authenticate token")
     return database.query(User).filter(User.username == payload.get("sub")).first()
 
 def create_access_token(data, expires_delta=None):
     to_encode = data.copy()
     if expires_delta:
         expire = datetime.utcnow() + expires_delta
     else:
-        expire = datetime.utcnow() + timedelta(minutes=15)
+        expire = datetime.utcnow() + timedelta(minutes=Constants.Auth.DEFAULT_EXPIRED_TIME)
```

Comment:
~~datetime.utcnow()~~
https://blog.miguelgrinberg.com/post/it-s-time-for-a-change-datetime-utcnow-is-now-deprecated
Use this (I think) instead
```python
datetime.now(timezone.utc)
```